src/Database/Schema/CollectionInterface.php

```diff
@@ -29,7 +29,7 @@ interface CollectionInterface
      *
      * @return array<string> The list of tables in the connected database/schema.
      */
-    public function listTables(): array;
+    public function listTables(array $options = []): array;
```

Comment:
Is the breaking change represented in the failing CI tests? I don't have MySQL 8 or MariaDB in my local, is there any easy way to simulate/debug changes in those environments?

This is my first time attempting to contribute back to any open source project so thank you for any guidance you can provide.

Reply:
>This is my first time attempting to contribute back to any open source project so thank you for any guidance you can provide.

That's great! I'm happy to help. :smile:

The breaking change is because we're adding parameters to an interface. Implementations of this interface outside of CakePHP will now raise fatal errors as the method signatures don't match. Adding a method will also fail in a similar way.

Instead what we could do is add another method to the implementations and use an `@method` comment in the class doc block. This will let IDEs know that the method exists. We'll also need to use `method_exists()` when calling the new function as static analysis tools will infer that the method doesn't exist :smile:

Reply:
Okay, I think I'm picking up what you're putting down. Either today or tomorrow I'll make a new commit to the PR and you can let me know if I got it.

To my other question, some of the CI tests are failing; in one case it's just a coding standard thing, easy enough to fix. But in other cases, like the "CI / testsuite (7.4, mariadb) (pull_request)" environment/task, it's some SQL error I can't make heads or tails of. Is there an easy way to test changes against these environments so I can get in with the debugger, or in this case, does it even matter, i.e. can I just not worry about that?

Thanks so much for the help!

Reply:
> But in other cases, like the "CI / testsuite (7.4, mariadb) (pull_request)" environment/task, it's some SQL error I can't make heads or tails of. Is there an easy way to test changes against these environments so I can get in with the debugger, or in this case, does it even matter, i.e. can I just not worry about that?

The failures could be related to this change but that seems unlikely. If the tests continue to fail with your next changes we can look into it. As for making a local environment you can use docker to pull down the images we use in CI and run tests against them.

Reply:
> 
Not totally sure I hit the mark on what you're looking for but take a look at the PR and let me know if I need to change things more.